third_party/move/move-compiler-v2/tests/checking/dotdot/dotdot_wrong_pos0.move

```diff
@@ -0,0 +1,10 @@
+module 0x42::test {
+    struct S {
+        x: bool,
+        y: u8
+    }
+
+    fun simple_0(x: S) {
```

Comment:
I think we are not supporting: `let x = S { y: 5, ..bar}` syntax available in Rust to construct one struct based on another. But we should have a test with such a syntax showing that it is not supported.

Reply:
Do we generally test for not supporting a feature (there are potentially infinitely of unimplemented features)? But we should mention this in the feature documentation.

Reply:
While there are potentially infinite number of unimplemented features, given how close we are to Rust in this syntax, this might be something a user might expect is supported, and is a very specific ask. So the test is to showcase that we indeed report a sane error in the case a user attempts to use this syntax, which is IMO quite likely if some one sees Move supports `..` in the contexts that we do.

Also, the PR description is sparse on describing all the contexts that `..` is supported - currently it needs to be mostly gleaned mostly from the test cases. Hence, my reasoning about why we should showcase which related features are supported and not supported through test cases.

Reply:
Done. Tested in `third_party/move/move-compiler-v2/tests/checking/dotdot/field_update_not_supported.move`.